docs/python-examples/room-assignment-transform.py

```
@@ -0,0 +1,80 @@
+if input_data and isinstance(input_data, dict) and 'elementSpaceMap' in input_data:
+    print("✓ Found room assignment data")
+    element_space_map = input_data['elementSpaceMap']
+    print(f"  Element-space map has {len(element_space_map)} entries")
+    
+    # The elementSpaceMap already uses GlobalIds as keys!
+    # We just need to filter for walls
+    
+    # Get wall GlobalIds from the IFC model
+    wall_global_ids = set()
+    
+    if model and hasattr(model, 'by_type'):
```

Ruff (0.12.2)

Undefined name `model`

(F821)